tests/test_smoketest.py

```diff
@@ -1,107 +1,50 @@
-
-def ENDC():
-    return "\033[0m" if sys.stdout.isatty() else ""
-
-
-def run(command, **kwargs):
-    print(GREEN(command))
-    try:
-        return subprocess.run(command, check=True, shell=isinstance(command, str), **kwargs)
-    except subprocess.CalledProcessError as e:
-        parser.exit(RED(f'{parser.prog}: Exit status {e.returncode} while running "{command}". Stopping.'))
-
-
-def run_for_json(command, **kwargs):
-    return json.loads(run(command, stdout=subprocess.PIPE, **kwargs).stdout.decode(sys.stdout.encoding))
-
 
 @testmode.integration
-class Smoketest(unittest.TestCase):
-    params = [
-        {
-            'starting_replica': Replica.aws,
-            'checkout_bucket': os.environ['DSS_S3_CHECKOUT_BUCKET'],
-            'test_bucket': os.environ['DSS_S3_BUCKET_TEST']
-        },
-        {
-            'starting_replica': Replica.gcp,
-            'checkout_bucket': os.environ['DSS_GS_CHECKOUT_BUCKET'],
-            'test_bucket': os.environ['DSS_GS_BUCKET_TEST']
-        }
-    ]
-    notification_bucket = os.environ['DSS_S3_BUCKET_TEST']
-
+class Smoketest(BaseSmokeTest):
+    # params = super().params
     @classmethod
     def setUpClass(cls):
-        cls.replicas = {param['starting_replica'] for param in cls.params}
-        if os.path.exists("dcp-cli"):
-            run("git pull --recurse-submodules", cwd="dcp-cli")
-        else:
-            run("git clone --depth 1 --recurse-submodules https://github.com/HumanCellAtlas/dcp-cli")
-        cls.workdir = tempfile.TemporaryDirectory(dir=os.getcwd(), prefix="smoketest-", suffix=".tmp")
-        if not args.clean:
-            # Disable workdir destructor
-            cls.workdir._finalizer.detach()  # type: ignore
-
-        # Create a virtualenv and install the CLI
-        #
-        venv = os.path.join(cls.workdir.name, "venv")
-        run(f"virtualenv -p {sys.executable} {venv}")
-        cls.venv_bin = os.path.join(venv, "bin", "")
-        run(f"{cls.venv_bin}pip install --upgrade .", cwd="dcp-cli")
-
-        # Configure the CLI
-        #
-        cli_config = {"DSSClient": {"swagger_url": f"https://{os.environ['API_DOMAIN_NAME']}/v1/swagger.json"}}
-        cli_config_filename = f"{cls.workdir.name}/cli_config.json"
-        with open(cli_config_filename, "w") as fh2:
-            fh2.write(json.dumps(cli_config))
-        os.environ["HCA_CONFIG_FILE"] = f"{cls.workdir.name}/cli_config.json"
+        super().setUpClass()
 
         # Prepare the bundle using stock metadata and random data
         #
         cls.bundle_dir = os.path.join(cls.workdir.name, "bundle")
-        shutil.copytree("tests/fixtures/datafiles/example_bundle", cls.bundle_dir)
+        shutil.copytree(os.environ["DSS_HOME"] + "/tests/fixtures/datafiles/example_bundle", cls.bundle_dir)
         with open(os.path.join(cls.bundle_dir, "async_copied_file"), "wb") as fh:
             fh.write(os.urandom(ASYNC_COPY_THRESHOLD + 1))
 
+    @classmethod
+    def tearDownClass(cls):
```

Comment:
Can this method be moved into the base class?

Reply:
changed.